apps/forms/control-value-accessor/src/app/rating-control/rating-control.component.ts

```diff
@@ -1,20 +1,52 @@
-import { Component, EventEmitter, Output } from '@angular/core';
+import { Component, forwardRef } from '@angular/core';
+import { ControlValueAccessor, NG_VALUE_ACCESSOR } from '@angular/forms';
 
 @Component({
   standalone: true,
   selector: 'app-rating-control',
   templateUrl: 'rating-control.component.html',
   styleUrls: ['rating-control.component.scss'],
+  providers: [
+    {
+      provide: NG_VALUE_ACCESSOR,
+      useExisting: forwardRef(() => RatingControlComponent),
+      multi: true,
+    },
+  ],
 })
-export class RatingControlComponent {
-  @Output()
-  readonly ratingUpdated: EventEmitter<string> = new EventEmitter<string>();
+export class RatingControlComponent implements ControlValueAccessor {
```

Comment:
You are missing the validation implementation

Reply:
Sorry, I didn't quite understand you. I added Validators.required to the rating control in the feedbackForm. So the control stays invalid until you click on it and the submit button is disabled. Do you want validation inside the RatingControlComponent ? Or do you want more validators like min, max to feedbackForm? Or maybe something else?

Reply:
Never mind. I didn't create that challenge and I thought you had to implement the `Validator` interface to validate your custom input. But all default validators are available when you implement a custom `ControlValueAccessor`. I've learnt something today. 👍

The interface 'Validators` is only useful if you want to add a custom validator.

Reply:
It would be cool to see your idea in your answer to this challenge 😉